API/hermes/hermes.cpp

```diff
@@ -1996,7 +1998,20 @@ void HermesRuntimeImpl::setPropertyValue(
 }
 
 bool HermesRuntimeImpl::isArray(const jsi::Object &obj) const {
-  return vm::vmisa<vm::JSArray>(phv(obj));
+  vm::JSObject *jsobj = static_cast<vm::JSObject *>(phv(obj).getPointer());
```

Comment:
We have an `isArray` in `Operations.h` that you can call into that will provide the right functionality here.

Reply:
Code updated.

Question: Should I call `runtime_.clearThrownValue()` on `ExecutionStatus::EXCEPTION` before returning `false`?

Reply:
You should use `checkStatus` to rethrow the exception as a C++ exception

Reply:
`checkStatus` isn't `const` thus cannot be used here without modification...

Reply:
It should be safe to remove the constness and call `checkStatus` here. Code updated.